transaction/transaction/src/main/java/io/helidon/transaction/Tx.java

```diff
@@ -0,0 +1,303 @@
+    @Target({ElementType.TYPE, ElementType.METHOD})
+    public @interface New { }
+
+    /**
+     * Defines {@link Type#NEVER} transaction type of the annotated method.
+     */
+    @Interception.Intercepted
+    @TransactionType(Type.NEVER)
+    @Retention(RetentionPolicy.SOURCE)
+    @Target({ElementType.TYPE, ElementType.METHOD})
+    public @interface Never { }
+
+    /**
+     * Defines {@link Type#REQUIRED} transaction type of the annotated method.
+     */
+    @Interception.Intercepted
+    @TransactionType(Type.REQUIRED)
+    @Retention(RetentionPolicy.SOURCE)
+    @Target({ElementType.TYPE, ElementType.METHOD})
+    public @interface Required { }
+
+    /**
+     * Defines {@link Type#SUPPORTED} transaction type of the annotated method.
+     */
+    @Interception.Intercepted
+    @TransactionType(Type.SUPPORTED)
+    @Retention(RetentionPolicy.SOURCE)
+    @Target({ElementType.TYPE, ElementType.METHOD})
+    public @interface Supported { }
+
+    /**
+     * Defines {@link Type#UNSUPPORTED} transaction type of the annotated method.
+     */
+    @Interception.Intercepted
+    @TransactionType(Type.UNSUPPORTED)
+    @Retention(RetentionPolicy.SOURCE)
+    @Target({ElementType.TYPE, ElementType.METHOD})
+    public @interface Unsupported { }
+
+    /**
+     * Execute provided task as database transaction.
+     * Transaction is handled automatically. Task computes and returns result.
+     *
+     * @param task task to run in transaction
+     * @param <T>  the result type of the task
+     * @return computed task result
+     * @throws TxException when result computation failed
+     */
+    public static <T> T transaction(Callable<T> task) {
+        return transaction(Type.REQUIRED, task);
+    }
+
+    /**
+     * Execute provided task as database transaction.
+     * Transaction is handled automatically. Task computes and returns result.
+     *
+     * @param type transaction type
+     * @param task task to run in transaction
+     * @param <T>  the result type of the task
+     * @return computed task result
+     * @throws TxException when result computation failed
+     */
+    public static <T> T transaction(Type type, Callable<T> task) {
+        return Services.get(TxSupport.class)
+                .transaction(type, task);
+    }
+
+    /**
+     * Execute provided task as database transaction.
+     * Transaction is handled automatically. Task does not return any result.
+     *
+     * @param task task to run in transaction
+     * @throws TxException when task computation failed
+     */
+    public static void transaction(Functions.CheckedRunnable<Exception> task) {
+        transaction(Type.REQUIRED, task);
+    }
+
+    /**
+     * Execute provided task as database transaction.
+     * Transaction is handled automatically. Task does not return any result.
+     *
+     * @param type transaction type
+     * @param task task to run in transaction
+     * @throws TxException when task computation failed
```

Comment:
Add `@throws NullPointerException` to conform to (implied) contract

Reply:
Not required here. NullPointerException will be cause of TxException if it's thrown.